Index: grub/grub-core/osdep/linux/getroot.c
===================================================================
--- grub.orig/grub-core/osdep/linux/getroot.c
+++ grub/grub-core/osdep/linux/getroot.c
@@ -21,6 +21,7 @@
 
 #include <sys/stat.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <assert.h>
 #include <fcntl.h>
 #include <unistd.h>
Index: grub/grub-core/osdep/unix/getroot.c
===================================================================
--- grub.orig/grub-core/osdep/unix/getroot.c
+++ grub/grub-core/osdep/unix/getroot.c
@@ -21,6 +21,7 @@
 
 #include <sys/stat.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <assert.h>
 #include <fcntl.h>
 #include <unistd.h>
